baybe/utils/validation.py

```diff
@@ -177,3 +177,21 @@ def validate_parameter_input(
                     f"the flag 'numerical_measurements_must_be_within_tolerance' "
                     f"to 'False' to disable this behavior."
                 )
+
+
+def validate_object_names(objects: Iterable[Parameter | Target]) -> None:
```

Comment:
I think good case for positional-only
```suggestion
def validate_object_names(objects: Iterable[Parameter | Target], /) -> None:
```